src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
Another idea, should this live as a helper instead?  the issue with original Dust is that it never had a good way of standardizing bodies. i'd like this explicit with lifecycles before we start teh slippery slope arguments.